schemas/v3.1/openapi-vocabulary.json

```diff
@@ -0,0 +1,1409 @@
+{
+  "$schema": "http://json-schema.org/draft/2019-WIP/schema#",
+  "$id": "https://spec.openapis.org/oas/3.1/schema",
+  "$vocabulary": {
+    "https://json-schema.org/draft/2019-WIP/vocab/core": true,
+    "https://json-schema.org/draft/2019-WIP/vocab/applicator": true,
+    "https://json-schema.org/draft/2019-WIP/vocab/validation": true,
+    "https://json-schema.org/draft/2019-WIP/vocab/meta-data": true,
+    "https://json-schema.org/draft/2019-WIP/vocab/format": true,
+    "https://json-schema.org/draft/2019-WIP/vocab/content": true,
+    "https://specs.openapis.org/oas/3.1/vocab/openapi": true
+  },
+  "$recursiveAnchor": true,
```

Comment:
This is not doing what you want it to do- the OAS file schema is not a recursive extension of a regular meta-schema.  It is a schema for a totally separate document format that embeds extended schemas.  Take a look at http://json-schema.org/work-in-progress/WIP-links.json for an example of how this sort of thing should work.

As noted in the comment on the vocabulary meta-schema you need three documents:

* A schema for the vocabulary (which can work as a meta-schema on its own, but is not very useful that way)
* A meta-schema for the OAS Schema Object (essentially the same as the standard JSON Schema `schema`, but with the OAS vocabulary added as the 7th entry in `$vocabularies`, and the OAS vocabulary schema as the 7th entry in the `allOf`)
* A schema for the OAS file format, which references the meta-schema for the OAS Schema Object

The OAS Schema Object meta-schema needs to be a standalone document because of how `$recursiveAnchor` and `$recursiveRef` work.  `$recursiveRef` can only target root schemas (entire schema documents) because otherwise it would be incredibly difficult to implement.

Reply:
Also, if we're moving to the new draft for the OAS meta-schema, I would strongly recommend starting from https://gist.github.com/handrews/6dfebd56ef97328f9e4dc7a47a1e8bc7 which is more concise, although not tested and not necessarily correct.